app/Services/DeliverableService.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace App\Services;
+
+use App\Resource;
+use App\Task;
+use App\Deliverable;
+use App\Contracts\DeliverableContract;
+
+class DeliverableService implements DeliverableContract
+{
+    public function getDataForViewingDeliverable($id)
+    {
+        $deliverable = Deliverable::find($id);
+        $tasks = $deliverable->getTasks()->get();
+        $show = false;
+        $taskId = null;
+        $data = [
+            'id' => $id,
+            'tasks' => $tasks,
+            'show' => $show,
+            'taskId' => $taskId,
+            'resourceName' => '',
+            'resourceTitle' => '',
+            'issues' => [],
+        ];
+
+        return $data;
+    }
+
+    public function getDataForViewingTask($did, $tid): array
+    {
+        $deliverable = Deliverable::find($did);
```

Comment:
Would it be better to inject the dependency of Deliverable into the service instead?

Reply:
No the service is the abstraction